.github/workflows/cibuildwheels.yml

```diff
@@ -98,10 +113,13 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install -r requirements-build.txt
+          # This should be in the [build-system] section of pyproject.toml
+          # python -m pip install -r requirements-build.txt
 
       - name: Build sdist
         run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements-build.txt
           python -m build --sdist
```

Comment:
The `python -m build` is sufficient. You can even replace all of the previous parts with `pipx build` (sets-up virtual environment, installs `build` and then runs `python -m build`).

Also you already have a previous workflow that executes `pytest`, consider consolidating these.